apps/server/src/infra/sync/media-licenses/service/vidis-sync.service.ts

```diff
@@ -0,0 +1,128 @@
+import { MediaSource } from '@modules/media-source';
+import { MediaSchoolLicenseService, MediaSchoolLicense, SchoolLicenseType } from '@modules/school-license';
+import { School, SchoolService } from '@modules/school';
+import { OfferDTO } from '@infra/vidis-client';
+import { Logger } from '@src/core/logger';
+import { ObjectId } from '@mikro-orm/mongodb';
+import { Injectable } from '@nestjs/common';
+import { SchoolForSchoolMediaLicenseSyncNotFoundLoggable } from '../loggable';
+
+@Injectable()
+export class VidisSyncService {
+	constructor(
+		private readonly mediaSchoolLicenseService: MediaSchoolLicenseService,
+		private readonly schoolService: SchoolService,
+		private readonly logger: Logger
+	) {}
+
+	public async syncMediaSchoolLicenses(mediaSource: MediaSource, vidisOfferItems: OfferDTO[]): Promise<void> {
+		const syncItemPromises: Promise<void>[] = vidisOfferItems.map(async (item: OfferDTO): Promise<void> => {
+			if (!item.schoolActivations || !item.offerId) {
+				return;
+			}
+
+			const mediumId = item.offerId.toString();
+
+			const officialSchoolNumbers = item.schoolActivations.map((activation) => this.removePrefix(activation));
+
+			let existingLicenses: MediaSchoolLicense[] = await this.mediaSchoolLicenseService.findAllByMediaSourceAndMediumId(
+				mediaSource.id,
+				mediumId
+			);
+
+			if (existingLicenses.length) {
+				existingLicenses = await this.removeNoLongerAvailableLicenses(existingLicenses, officialSchoolNumbers);
+			}
+
+			const existingLicenseSchoolNumberSet = new Set(
```

Comment:
I think this would be more readable if we build a more structured delta, like:
```
const licensesToDelete = [];
const existingLicenseSchoolNumbers = new Set();

existingLicenses.forEach((license) => {
	if(!license.school.officialSchoolNumber || !vidisSchoolNumberSet.has(license.school.officialSchoolNumber)) {
		licensesToDelete.push(license);
	} else {
		existingLicenseSchoolNumbers.push(license.school.officialSchoolNumber);
	}
})

await this.deleteLicenses(licensesToDelete);
await this.createLicenses(mediumId, vidisSchoolNumbers, existingLicenseSchoolNumbers);
```
and moving the create logic into its own private method makes it more readable too.